static/app/views/dashboardsV2/widgetBuilder/yAxisSelector.tsx

```diff
@@ -0,0 +1,261 @@
+};
+
+function DeleteButton({onDelete}) {
+  return (
+    <Button
+      size="zero"
+      borderless
+      onClick={onDelete}
+      icon={<IconDelete />}
+      title={t('Remove this Y-Axis')}
+      aria-label={t('Remove this Y-Axis')}
+    />
+  );
+}
+
+function AddButton({title, onAdd}) {
+  return (
+    <Button size="small" aria-label={title} onClick={onAdd} icon={<IconAdd isCircled />}>
+      {title}
+    </Button>
+  );
+}
+
+function YAxisSelector({
+  displayType,
+  fields,
+  style,
+  fieldOptions,
+  onChange,
+  errors,
+}: Props) {
+  const organization = useOrganization();
+  // const isMetricWidget = widgetType === WidgetType.METRICS;
+
+  function handleAdd(event: React.MouseEvent) {
+    event.preventDefault();
+
+    const newFields = [
+      ...fields,
+      {kind: FieldValueKind.FIELD, field: ''} as QueryFieldValue,
+    ];
+    onChange(newFields);
+  }
+
+  function handleAddEquation(event: React.MouseEvent) {
+    event.preventDefault();
+
+    const newFields = [
+      ...fields,
+      {kind: FieldValueKind.EQUATION, field: ''} as QueryFieldValue,
+    ];
+    onChange(newFields);
+  }
+
+  function handleRemove(event: React.MouseEvent, fieldIndex: number) {
+    event.preventDefault();
+
+    const newFields = [...fields];
+    newFields.splice(fieldIndex, 1);
+    onChange(newFields);
+  }
+
+  function handleChangeField(value: QueryFieldValue, fieldIndex: number) {
+    const newFields = [...fields];
+    newFields[fieldIndex] = value;
+    onChange(newFields);
+  }
+
+  function handleTopNChangeField(value: QueryFieldValue) {
+    const newFields = [...fields];
+    newFields[fields.length - 1] = value;
+    onChange(newFields);
+  }
+
+  // Any function/field choice for Big Number widgets is legal since the
+  // data source is from an endpoint that is not timeseries-based.
+  // The function/field choice for World Map widget will need to be numeric-like.
+  // Column builder for Table widget is already handled above.
+  const doNotValidateYAxis = displayType === 'big_number';
+
+  function filterPrimaryOptions(option) {
+    // Only validate function names for timeseries widgets and
+    // world map widgets.
+    if (!doNotValidateYAxis && option.value.kind === FieldValueKind.FUNCTION) {
+      const primaryOutput = aggregateFunctionOutputType(
+        option.value.meta.name,
+        undefined
+      );
+      if (primaryOutput) {
+        // If a function returns a specific type, then validate it.
+        return isLegalYAxisType(primaryOutput);
+      }
+    }
+
+    // if (
+    //   widgetType === WidgetType.METRICS &&
+    //   (displayType === DisplayType.TABLE || displayType === DisplayType.TOP_N)
+    // ) {
+    //   return (
+    //     option.value.kind === FieldValueKind.FUNCTION ||
+    //     option.value.kind === FieldValueKind.TAG
+    //   );
+    // }
+
+    return option.value.kind === FieldValueKind.FUNCTION;
+  }
+
+  const filterAggregateParameters = fieldValue => option => {
```

Comment:
can we add types here?

maybe we could write something like

```suggestion
  function filterAggregateParameters(fieldValue: QueryFieldValue) {
    return (option: FieldValueOption) => {
    ...
```

Reply:
I added types, but I was getting a TypeError so I came up with [this fix to handle the extra keys (set and counter, from metrics) separately](https://github.com/getsentry/sentry/pull/31915/files#diff-fe9da67febb38b6f64afaa0cc75fba57c658e8abe8f9361f5b88175394d12f8cR130)

I also have an equivalent PR to add this change to the current query fields #31970